### PR TITLE
Single Hacker View

### LINKED ISSUE #254: Single Hacker View Modal
Admins should be able to get a detailed view of a single hacker showing all of their information. Components should be able to be re-used from the application and account creation pages.

### PR DESCRIPTION
**Pull request by erickzhao**
Closes #254 

### What this does
Adds a single hacker modal in the admin view. Allows a searcher to see a hacker's full application, as well as change their update status. Can toggle between hackers in modal view (with buttons or keyboard controls).

### Demo
screencast 2019-01-14 18-36-11

### Considerations

* @krubenok what are the exact requirements for someone to be able to change status? Is it just having the Staff role?
~* The previous/next feature is missing for now.~

### CODE REVIEW

**Comment by pierreTklein** [comment]
If the account was deleted, we get this error:
image

**Comment by pierreTklein** [comment]
Modal looks great!

image

File: src/Nav/Nav.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ interface INavProps {
 }
 
 export const Nav = styled.nav<INavProps>`
-  z-index: 11;
+  z-index: 2;
```
Comment (pierreTklein) [inline src/Nav/Nav.tsx]: Does this fuck up any other relative indices

File: src/Search/SingleHackerModal.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  public render() {
+    return (
+      <div>
+        <Button type="button" onClick={this.handleOpenModal}>
+          See more
+        </Button>
+        <Modal
+          appElement={document.getElementById('root') || undefined}
+          style={{ overlay: { zIndex: 3 } }}
```
Comment (pierreTklein) [inline src/Search/SingleHackerModal.tsx]: Can we make this a `styled-component`?

File: src/Search/SingleHackerModal.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          isOpen={this.state.showModal}
+          contentLabel="Example Modal"
```
Comment (pierreTklein) [inline src/Search/SingleHackerModal.tsx]: "Example Modal" -> "Single Hacker View"?

File: src/Search/SingleHackerView.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+      canEdit: true,
+      isLoading: false,
+    };
+  }
+
+  public componentDidMount() {
+    const {
+      hacker: { status },
+    } = this.props;
+    this.setState({ status });
```
Comment (pierreTklein) [inline src/Search/SingleHackerView.tsx]: We can move this logic to the constructor, when we do `this.state = {...}`

File: src/Search/SingleHackerView.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    this.submit = this.submit.bind(this);
+  }
+
+  public render() {
+    const { hacker } = this.props;
+    const account = (hacker.accountId as IAccount) || {};
+    return (
+      <article style={{ maxWidth: '800px', margin: '0 auto' }}>
```
Comment (pierreTklein) [inline src/Search/SingleHackerView.tsx]: Should we make a `styled-component` out of this?

File: src/Search/SingleHackerView.tsx [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+  private async submit() {
+    try {
+      const { hacker } = this.props;
+      const { status } = this.state;
+      this.setState({ isLoading: true });
+      await Hacker.updateStatus(hacker.id, status);
+      this.setState({ isLoading: false });
+      toast.success(`Hacker status updated to ${status}!`);
+    } catch (e) {
+      toast.error(`Hacker status update failed! Message: ${e}`);
```
Comment (pierreTklein) [inline src/Search/SingleHackerView.tsx]: Could use `ValidationErrorGenerator` here, if you want

File: src/Search/SingleHackerView.tsx [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+          <Flex
+            width="100%"
+            flexWrap="wrap"
+            justifyContent="space-between"
+            alignItems="center"
+          >
+            <Box width={[1, 1 / 2]}>
+              <strong>GitHub</strong>:{' '}
+              {hacker.application.portfolioURL.github ? (
+                <a href={hacker.application.portfolioURL.github}>
```
Comment (pierreTklein) [inline src/Search/SingleHackerView.tsx]: Can we add `target="_blank"` so that these links open up into a new tab?

Comment (pierreTklein) [reply src/Search/SingleHackerView.tsx]: Could we refactor this into a `LinkView` component?

File: src/Nav/Nav.tsx (hunk 1 above)
Comment (erickzhao) [reply src/Nav/Nav.tsx]: Oh yeah it does. We have a bunch set to `10`. I'll fix that.

File: src/Search/SingleHackerModal.tsx (hunk 2 above)
Comment (erickzhao) [reply src/Search/SingleHackerModal.tsx]: https://github.com/reactjs/react-modal/issues/603#issuecomment-378847885

I was having issues before but it seems like there's a workaround for it. I'll try plugging that in and getting back to you.

File: src/Search/SingleHackerView.tsx (hunk 5 above)
Comment (erickzhao) [reply src/Search/SingleHackerView.tsx]: just refactored the styles out lol

File: src/Search/SingleHackerView.tsx (hunk 7 above)
Comment (erickzhao) [reply src/Search/SingleHackerView.tsx]: What's a `LinkView`?

**Comment by pierreTklein** [comment]
Are we good on a merge, or are we waiting?

File: src/Search/SingleHackerView.tsx [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+              <Box width={[1, 1 / 2]}>
+                <StyledSelect
+                  isTight={true}
+                  className="react-select-container"
+                  classNamePrefix="react-select"
+                  options={getOptionsFromEnum(HackerStatus)}
+                  isDisabled={!this.state.canEdit}
+                  onChange={this.handleChange}
+                  value={{
+                    label: this.state.status,
```
Comment (pierreTklein) [inline src/Search/SingleHackerView.tsx]: This does not change when user is switching between hackers

**Pull request merged**